server/server_test.go

```diff
@@ -35,6 +35,86 @@ func (lb *testLoadBalancer) Servers() []*url.URL {
 	return []*url.URL{}
 }
 
+func TestPrepareServerTimeouts(t *testing.T) {
+	tests := []struct {
+		desc             string
+		globalConfig     GlobalConfiguration
+		wantIdleTimeout  time.Duration
+		wantReadTimeout  time.Duration
+		wantWriteTimeout time.Duration
+	}{
+		{
+			desc: "full configuration",
+			globalConfig: GlobalConfiguration{
+				RespondingTimeouts: &RespondingTimeouts{
+					IdleTimeout:  flaeg.Duration(10 * time.Second),
+					ReadTimeout:  flaeg.Duration(10 * time.Second),
+					WriteTimeout: flaeg.Duration(10 * time.Second),
```

Comment:
I'd probably choose slightly different numbers for the three timeouts just to rule out the possibility that timeout values are inadvertently reused.